Classes/Decoder/UrlDecoder.php

```diff
@@ -1225,7 +1204,7 @@ protected function searchForPathOverrideInPagesLanguageverlay($path) {
 				$result = GeneralUtility::makeInstance('DmitryDulepov\\Realurl\\Cache\\PathCacheEntry');
 				/** @var \DmitryDulepov\Realurl\Cache\PathCacheEntry $result */
 				$result->setLanguageId($this->detectedLanguageId);
-				$result->setPageId((int)$row['pid']);
+				$result->setPageId((int)$row['uid']);
```

Comment:
Integrated manually to the current development. Unfortunately I cannot take the rest because it removes functionality that either was there traditionally or made for future use.